btk/metrics.py

```diff
@@ -1,484 +1,407 @@
+            meas_func = measure_results.keys()
+            metrics_results = {}
+            for f in meas_func:
+                metrics_results_f = compute_metrics(
+                    blend_results["blend_images"],
+                    blend_results["isolated_images"],
+                    blend_results["blend_list"],
+                    measure_results[f]["catalog"],
+                    measure_results[f]["segmentation"],
+                    measure_results[f]["deblended_images"],
+                    self.use_metrics,
+                    self.meas_band_num,
+                    self.target_meas,
+                    blend_id_start=self.blend_counter,
+                )
+                metrics_results[f] = metrics_results_f
+
+        else:
+            metrics_results = compute_metrics(
+                blend_results["blend_images"],
+                blend_results["isolated_images"],
+                blend_results["blend_list"],
+                measure_results["catalog"],
+                measure_results["segmentation"],
+                measure_results["deblended_images"],
+                self.use_metrics,
+                self.meas_band_num,
+                self.target_meas,
+                blend_id_start=self.blend_counter,
             )
-        true_table, detected_table, detection_summary = evaluate_detection(
-            batch_detection_result[0], batch_detection_result[1], batch_index=i
-        )
-        results["detection"][0] = astropy.table.vstack([results["detection"][0], true_table])
-        results["detection"][1] = astropy.table.vstack([results["detection"][1], detected_table])
-        results["detection"][2].extend(detection_summary)
-        # Evaluate segmentation algorithm
-        segmentation = metrics_params.get_segmentation()
-        results["segmentation"].append(evaluate_segmentation(segmentation, index=i))
-        # Evaluate flux measurement algorithm
-        flux = metrics_params.get_flux()
-        results["flux"].append(evaluate_flux(flux, index=i))
-        # Evaluate shape measurement algorithm
-        shapes = metrics_params.get_shapes()
-        results["shapes"].append(evaluate_shapes(shapes, index=i))
-    return results
+
+        self.blend_counter += len(blend_results["blend_list"])
+        return blend_results, measure_results, metrics_results
+
+
+def run_metrics(metrics_generator: MetricsGenerator, n_batches=100):
+    """Uses a `metrics_generator` objec to summarize metrics results for `n_batches` batches."""
+    measure_funcs = metrics_generator.measure_generator.measure_functions
+    summary_tables = {f: astropy.table.Table() for f in measure_funcs}
+    for i in range(n_batches):
+        blend_results, measure_results, metrics_results = next(metrics_generator)
+        for f in measure_funcs:
+            summary_tables[f] = astropy.table.vstack(
+                summary_tables[f], metrics_results["galaxy_summary"]
+            )
+    return summary_tables
```

Comment:
Here is a first draft of a function that uses info from multiple batches. I would also like to add overall results on all batches/blends. I guess it depends how we want to display the data...

Reply:
Definitely open to suggestions on this one, still thinking on how to summarize this in the best way